ledger/metrics/src/main/scala/com/daml/telemetry/TelemetryContext.scala

```diff
@@ -117,11 +117,14 @@ protected class DefaultTelemetryContext(protected val tracer: Tracer, protected
   ): T = {
     val subSpan = createSubSpan(spanName, kind, attributes: _*)
 
-    try {
-      body(DefaultTelemetryContext(tracer, subSpan))
-    } finally {
-      subSpan.end()
+    val result = Try { body(DefaultTelemetryContext(tracer, subSpan)) }
+    result match {
+      case Failure(exception) => subSpan.recordException(exception)
+      case Success(_) =>
     }
+
+    subSpan.end()
+    result.get
```

Comment:
I think I preferred the `try` syntax a little more, but I don't particularly mind this version either.

```suggestion
    try {
      body(DefaultTelemetryContext(tracer, subSpan))
    } catch (exception) {
      subSpan.recordException(exception)
      throw exception
    } finally {
      subSpan.end()
    }
```